### PR TITLE
User/lmh/fv gfs

### PR DESCRIPTION
**Pull request by lharris4**
This includes Zhi's updates for the new nesting capabilities. I don't know if Zhi pushed his changes to the official FMS repo or even if he has authorized such a thing.

Lucas

### CODE REVIEW

**Comment by wrongkindofdoctor** [comment]
None of Zhi's changes to the mpp_domains files have been pushed to the GFDL master because Ray had problems compiling with them. Zhi made the changes on a branch that was out of sync with the main I/O development branch, so FMS will test the domains once we have worked through the I/O.

**Comment by lharris4** [comment]
Hi, Jess. Thanks. There are updates to the dynamical core that are
necessary for Zhi's changes to work. I am preparing a "sanitized" version
of the core for transmission outside of GFDL that will include the changes
in FV3 to support the FMS changes but that will not include codes that
should not be made public yet.

Lucas

File: mpp/mpp_domains.F90 [hunk 1]
```diff
@@ -442,10 +450,20 @@ module mpp_domains_mod
 
   end type nestSpec
 
-
-
   type nest_domain_type
+     character(len=NAME_LENGTH)     :: name
```
Comment (underwoo) [inline mpp/mpp_domains.F90]: Where is the documentation for the nest_domain_type and nest_level_type?  Documentation of the derived types is something we should _always_ have.

File: mpp/include/mpp_domains_define.inc [hunk 2]
```diff
@@ -1347,6 +1349,9 @@ end subroutine check_message_size
        end if
     end do
 
+    allocate(domain%tile_id_all(num_tile))
```
Comment (underwoo) [inline mpp/include/mpp_domains_define.inc]: This is a comment about code not touched:  The `tile_id_local` allocatable array is not deallocated in this routine.  While most compilers will, possibly, automatically deallocate it is much safer to add a deallocate in the routine.  I'm not going to have this hold up the merge, but it is something that should be addressed.

File: mpp/include/mpp_domains_misc.inc [hunk 3]
```diff
@@ -512,7 +512,7 @@ end subroutine init_nonblock_type
       logical :: native         !true if I'm on the pelist of this domain
       integer :: listsize, listpos
       integer :: n
-      integer, dimension(11) :: msg, info         !pe and compute domain of each item in list
+      integer, dimension(12) :: msg, info         !pe and compute domain of each item in list
```
Comment (underwoo) [inline mpp/include/mpp_domains_misc.inc]: I do not like this `12` (previously 11) to be coded like this.  It does not allow future developers to know why the dimension of these arrays is specifically 12.

File: mpp/include/mpp_domains_misc.inc [hunk 4]
```diff
@@ -571,12 +571,13 @@ end subroutine init_nonblock_type
       else
          info(11) = 0
       endif
+      info(12) = domain%ntiles
```
Comment (underwoo) [inline mpp/include/mpp_domains_misc.inc]: I am concerned if the `info(11) = 0` on line 572 should be `info(12) = 0`.  Without knowing why info has 12 elements, I am unable to determine the reason.

File: mpp/include/mpp_domains_misc.inc [hunk 5]
```diff
@@ -571,12 +571,13 @@ end subroutine init_nonblock_type
...
 !broadcast your info across current pelist and unpack if needed
       listpos = 0
       do n = 0,mpp_npes()-1
          msg = info
          if( mpp_pe().EQ.pes(n) .AND. debug )write( errunit,* )'PE ', mpp_pe(), 'broadcasting msg ', msg
-         call mpp_broadcast( msg, 11, pes(n) )
+         call mpp_broadcast( msg, 12, pes(n) )
```
Comment (underwoo) [inline mpp/include/mpp_domains_misc.inc]: Here is that `12` again.

File: mpp/include/mpp_domains_misc.inc [hunk 6]
```diff
@@ -733,6 +735,239 @@ end subroutine init_nonblock_type
...
 
+    subroutine mpp_broadcast_domain_nest_fine( domain, tile_nest )
+!broadcast domain (useful only outside the context of its own pelist)
+      type(domain2D), intent(inout) :: domain
+      integer,        intent(in)    :: tile_nest(:)
+      integer, allocatable :: pes(:)
+      logical :: native         !true if I'm on the pelist of this domain
+      integer :: listsize, listpos, nestsize(size(tile_nest(:)))
+      integer :: n, tile, ind, num_nest
+      integer, dimension(15) :: msg, info         !pe and compute domain of each item in list
```
Comment (underwoo) [inline mpp/include/mpp_domains_misc.inc]: And now we have `15` elements.

File: mpp/mpp_domains.F90 [hunk 7]
```diff
@@ -319,6 +325,7 @@ module mpp_domains_mod
   type domain1D_spec
      private
      type(domain_axis_spec) :: compute
+     type(domain_axis_spec) :: global
```
Comment (underwoo) [inline mpp/mpp_domains.F90]: derived type documentation?

File: sat_vapor_pres/sat_vapor_pres.F90 [hunk 8]
```diff
@@ -535,7 +535,7 @@ module sat_vapor_pres_mod
 
 ! The default values below preserve the behavior of omsk and earlier revisions.
  logical :: show_bad_value_count_by_slice=.true.
- logical :: show_all_bad_values=.false.
+ logical :: show_all_bad_values=.true. ! Changed LMH 22may2018
```
Comment (underwoo) [inline sat_vapor_pres/sat_vapor_pres.F90]: A change in default behavior should be included in the release notes.

Comment (bensonr) [inline sat_vapor_pres/sat_vapor_pres.F90]: Is there universal acceptance for changing the default behavior of a namelist controlled variable?

**Pull request closed without merging**